File: java/logic-2/closeFar.java
/* Given three ints, a b c, return true if one of b or c is "close" 
 * (differing from a by at most 1), while the other is "far", differing from 
 * both other values by 2 or more. Note: Math.abs(num) computes the absolute 
 * value of a number.
 */
public boolean closeFar(int a, int b, int c) {
    return (isClose(a, b) && isFar(a, b, c)) ||
        (isClose(a, c) && isFar(a, c, b));
}

public boolean isClose(int a, int b) {
    return Math.abs(a - b) <= 1;
}

public boolean isFar(int a, int b, int c) {
    return Math.abs(a - c) >= 2 && Math.abs(b - c) >= 2;
}

File: python/logic-2/close_far.py
# Given three ints, a b c, return True if one of b or c is "close" (differing 
# from a by at most 1), while the other is "far", differing from both other 
# values by 2 or more.
def close_far(a, b, c):
    return (is_close(a, b) and is_far(a, b, c)) or \ 
        (is_close(a, c) and is_far(a, c, b))
        
def is_close(a, b):
    return abs(a - b) <= 1
    
def is_far(a, b, c):
    return abs(a - c) >= 2 and abs(b - c) >= 2
